### PR TITLE
fix invalid escape warnings #162

### PR DESCRIPTION
**Pull request by bebuch**
The number of warnings from pytest drops from 22 to 6 with this patch. I applied raw strings to all label-like stuff, not just where necessary. This should make it more likely that raw strings will always be used on copy & paste in the future. ;-)

### CODE REVIEW

**Comment by fariza** [comment]
@zhaoyutong43210 please review

File: emva1288/process/plotting.py [hunk 1]
```diff
@@ -370,9 +370,9 @@ def plot(self, test):
 class PlotHorizontalSpectrogramPRNU(Emva1288Plot):
     '''Create Horizontal spectrogram PRNU plot'''
 
-    name = 'Horizontal spectrogram PRNU'
-    xlabel = 'cycles [periods/pixel]'
-    ylabel = 'Standard deviation and\nrelative presence of each cycle [%]'
+    name = r'Horizontal spectrogram PRNU'
+    xlabel = r'cycles [periods/pixel]'
+    ylabel = r'Standard deviation and\nrelative presence of each cycle [%]'
```
[the remark this answers is not in the document]
Comment (zhaoyutong43210) [reply emva1288/process/plotting.py:375]: everything here looks good.

**Comment by zhaoyutong43210** [comment]
I got an error while generating the report.tex, it turns out it's missing a newline ` \\` after the `& % $DR_{dB}$ &`, I will provide a solution soon.

**Comment by zhaoyutong43210** [comment]
please change line 109 in emva1288/report/templates/op.tex
from 
`%{{ op.results.sensitivity.DR_dB.unit %}}[3mm]`
to
`%{{ op.results.sensitivity.DR_dB.unit %}} \\[3mm]`
this is a bug I should be responsible for, but this would fix this.

File: emva1288/process/plotting.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (bebuch) [reply emva1288/process/plotting.py:375]: Great, thanks a lot for checking!

**Review by zhaoyutong43210 — APPROVED** [review]

**Comment by bebuch** [comment]
@zhaoyutong43210 Can you please describe me briefly how to create the report? Haven't found it on my own. Then I can test it next time or even better I can provide a unittest to find at least such syntax errors directly. ;-)

**Comment by bebuch** [comment]
I removed the commit with the ` %{{ op.results.sensitivity.DR_dB.unit %}} \\[3mm]` because it is also included in #167. This way we don't get into merge conflicts.

**Comment by fariza** [comment]
@zhaoyutong43210 @aturcotte what do you think?

**Comment by zhaoyutong43210** [comment]
> @zhaoyutong43210 @aturcotte what do you think?

I think this is okay to merge. I have checked the appearance of the report and everything seems okay, and indeed the number of warnings has decreased.

**Review by zhaoyutong43210 — DISMISSED** [review]

**Comment by aturcotte** [comment]
I am encountering the following error when running `pdflatex report.tex`:
image

File: emva1288/process/plotting.py [hunk 2]
```diff
@@ -131,20 +131,20 @@ def reduce_ticks(self, ax, axis, n=4):
 
 
 class PlotSensitivity(Emva1288Plot):
-    name = 'Sensitivity'
-    xlabel = '$\mu_p$ [mean number of photons/pixel]'
-    ylabel = '$\mu_y - \mu_{y.dark}$ [DN]'
+    name = r'Sensitivity'
```
Comment (aturcotte) [inline emva1288/process/plotting.py:134]: Converting this to a raw string, along with all other strings that do not contain backslashes, is unnecessary.

File: emva1288/process/plotting.py [hunk 3]
```diff
@@ -168,12 +168,12 @@ def rearrange(self):
 
 
 class PlotUyDark(Emva1288Plot):
-    '''Create $\mu_{y.dark}$ plot'''
+    '''Create $\\mu_{y.dark}$ plot'''
```
Comment (aturcotte) [inline emva1288/process/plotting.py:171]: Alternatively, you could use a raw string here.
```suggestion
    r'''Create $\mu_{y.dark}$ plot'''
```

File: emva1288/process/plotting.py (hunk 1 above)
Comment (aturcotte) [inline emva1288/process/plotting.py:375]: We want `\n` to actually be treated as a newline character here.
```suggestion
    ylabel = 'Standard deviation and\nrelative presence of each cycle [%]'
```

File: emva1288/process/plotting.py (hunk 3 above)
Comment (bebuch) [reply emva1288/process/plotting.py:171]: I wasn't sure which was better at that point. Since you also suggest it, I'll take it ;-)

**Comment by bebuch** [comment]
> I am encountering the following error when running `pdflatex report.tex`: image

That's an regression from [e6e9b836d](https://github.com/EMVA1288/emva1288/commit/e6e9b836d7656719c51b7679209a0c740d98be72#diff-539a798adb488b7696f464c0a3d16443a1df0371c1afa375bb355c284d930371R109). It will get fixed by #167. See my note [1249112384](https://github.com/EMVA1288/emva1288/pull/163#issuecomment-1249112384).

File: emva1288/process/plotting.py (hunk 1 above)
Comment (aturcotte) [reply emva1288/process/plotting.py:375]: As you can see, the newline is not treated correctly with your proposed change.
image

**Comment by aturcotte** [comment]
> @zhaoyutong43210 @aturcotte what do you think?



> > I am encountering the following error when running `pdflatex report.tex`: image
> 
> That's an regression from [e6e9b836d](https://github.com/EMVA1288/emva1288/commit/e6e9b836d7656719c51b7679209a0c740d98be72#diff-539a798adb488b7696f464c0a3d16443a1df0371c1afa375bb355c284d930371R109). It will get fixed by #167. See my note [1249112384](https://github.com/EMVA1288/emva1288/pull/163#issuecomment-1249112384).

Thanks for the reminder. I made a small change to get it working so I could verify the output.

File: emva1288/process/plotting.py (hunk 2 above)
Comment (aturcotte) [reply emva1288/process/plotting.py:134]: Personally, I think only using them when necessary makes it clearer _why_ they are necessary. Applying this everywhere, it seems more likely that we'll end up with an untreated `\n`.

@fariza Any thoughts on this?

Comment (bebuch) [reply emva1288/process/plotting.py:134]: Sounds like a strong argument to me. :-)

**Comment by bebuch** [comment]
@aturcotte @zhaoyutong43210 

According to our discussion, raw strings are now used only where technically necessary.

I also noticed that the tests are complaining about quite a few doc strings in `results.py`. Python apparently has a convention for this, which I have applied accordingly.

- https://peps.python.org/pep-0257/

Now all `DeprecationWarning: invalid escape sequence` warnings are gone from the tests. :-)

File: emva1288/process/plotting.py [hunk 4]
```diff
@@ -454,17 +454,17 @@ def plot(self, test):
 
         ax.plot((routines.GetFrecs(spectrogram)),
                 (np.sqrt(spectrogram)),
-                label='Data',
+                label=r'Data',
```
Comment (aturcotte) [inline emva1288/process/plotting.py:457]:
```suggestion
                label='Data',
```

File: emva1288/process/plotting.py [hunk 5]
```diff
@@ -454,17 +454,17 @@ def plot(self, test):
...
                 gid='%d:data' % test.id)
 
         ax.axhline(test.PRNU1288,
-                   label='$PRNU_{1288}$',
+                   label=r'$PRNU_{1288}$',
```
Comment (aturcotte) [inline emva1288/process/plotting.py:461]:
```suggestion
                   label='$PRNU_{1288}$',
```

File: emva1288/process/plotting.py [hunk 6]
```diff
@@ -492,13 +492,13 @@ def plot(self, test):
                 gid='%d:data' % test.id)
 
         ax.axhline(test.DSNU1288_DN(),
-                   label='$DSNU_{1288.DN}$',
+                   label=r'$DSNU_{1288.DN}$',
```
Comment (aturcotte) [inline emva1288/process/plotting.py:495]:
```suggestion
                   label='$DSNU_{1288.DN}$',
```

**Comment by bebuch** [comment]
Thanks @aturcotte!

**Review by aturcotte — APPROVED** [review]

**Comment by fariza** [comment]
@zhaoyutong43210 please do a last pass and test again from your side

**Comment by zhaoyutong43210** [comment]
Just tested, and everything looks good on my side

**Review by zhaoyutong43210 — APPROVED** [review]

**Pull request merged**